operator/internal/controller/redpanda/redpanda_node_pvc_controller.go

```diff
@@ -124,46 +117,38 @@ func (r *RedpandaNodePVCReconciler) reconcile(ctx context.Context, req ctrl.Requ
 			errs = errors.Join(errs, fmt.Errorf("worker node annotation not found or node name is empty: %q", workerNode)) //nolint:goerr113 // joining since we do not error here
 		}
 
-		// we are not being deleted, move on
+		// K8S Node label from PVC does not matches delete K8S Node event
```

Comment:
This comment says what the below check is but it doesn't say _why_ this check is being performed.